### PR TITLE
Remove unnecessary errors from missing jetpack version

### PR DESCRIPTION
**Pull request by planarvoid**
I've noticed that we were having an exception in the logs that was saying that Jetpack version is illegal. This is happening on my site when the jetpack version is "false". Checking the code that produces this exception there is a very weird behaviour that catches the exception and returns "true" (meaning that the site has the correct jetpack version) when the version parsing fails. I've changed this behaviour because I'm pretty sure it's wrong.
I've added unit tests to test the changed behaviour.

To test:
* Test the app with a self-hosted site (jurassic.ninja with and without Jetpack)
* Check that the plugins item is visible only when it should be
* Check that there is no error in the Logcat

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/util/SiteUtilsTest.kt [hunk 1]
```diff
@@ -45,4 +46,72 @@ class SiteUtilsTest {
...
+        site.origin = 1
+        site.setIsJetpackConnected(true)
+
+        val hasMinimalJetpackVersion = SiteUtils.checkMinimalJetpackVersion(site, "5.8")
+
+        assertThat(hasMinimalJetpackVersion).isTrue()
+    }
+
+    @Test
+    fun `checkMinimalJetpackVersion returns false when version is lower than to input`() {
```
Comment (develric) [inline WordPress/src/test/java/org/wordpress/android/util/SiteUtilsTest.kt]: super minor np; I think the to in `lower than (to) input` could be a copy/paste typo? :)

File: WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java [hunk 2]
```diff
@@ -268,14 +268,18 @@ public static boolean checkMinimalJetpackVersion(SiteModel site, String limitVer
                 if (index > 0) {
                     jetpackVersion = jetpackVersion.substring(0, index);
                 }
+                // Jetpack version is false on self-hosted sites
```
Comment (develric) [inline WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java]: Maybe we could consider to elaborate more on this comment, wdyt?

**Comment by planarvoid** [comment]
@develric thanks for the review! If I remember correctly, the condition for the `Lazy load` is different than for `Site accelerator` and you can actually have the `Lazy load` on any site. It's just duplicated so it's also nested in the `Site accelerator`. There is no condition in the code to hide the `Lazy load` option.

**Comment by oguzkocer** [comment]
This issue is targeting the current frozen milestone which is about to be released. I am bumping it to `14.6` which is when a PR that lands to `develop` in the next 2 weeks will be released. If it's more urgent than that, please make sure to target `release/14.5` branch which will be available later today and update the milestone for the issue accordingly.

**Pull request merged**